fvm/runtime/reusable_cadence_runtime.go

```diff
@@ -149,8 +149,9 @@ func (reusable *ReusableCadenceRuntime) ExecuteScript(
 	return reusable.Runtime.ExecuteScript(
 		script,
 		runtime.Context{
-			Interface: reusable.fvmEnv,
-			Location:  location,
+			Interface:   reusable.fvmEnv,
+			Location:    location,
+			Environment: reusable.Environment,
```

Comment:
cc: @janezpodhostnik

Reply:
Interesting that this was missing.

Reply:
When executing scripts, providing an environment which has the EVM functionality enabled is correct and necessary.

However, it matters what _kind_ of environment is passed: In general, scripts have access to additional functions which are not defined in transactions, for example `getAuthAccount`. For this purpose Cadence provides two environment constructors:
- `runtime.NewBaseInterpreterEnvironment` for e.g. transactions
- `runtime.NewScriptInterpreterEnvironment` for scripts, which has e.g. `getAuthAccount` defined

Passing an environment is an opt-in optimization, so if no environment is passed, Cadence instantiates a new one each time. If an environment is passed in the config, it is re-configured/re-used.

Passing `reusable.Environment` here is wrong, because it is an environment for transactions, not one sufficient for scripts.

To fix this, as far as I understand FVM, you want to do exactly the same as what is done to create `reusable.Environment`, but use `runtime.NewScriptInterpreterEnvironment` for it.

It would also be good to add a test case to FVM itself that simply calls `getAuthAccount` in a script to prevent regressions like this one.

Reply:
Thanks for the detailed explanation Bastian, @janezpodhostnik is this something you can help with by creating a PR to this one?

Reply:
updated to use separate runtime environments

Reply:
a05a40c looks good! I'll try updating the `evm_test.go` tomorrow and see if this works as expected